index.js

```diff
@@ -237,74 +244,80 @@ function calcChecksumAsIndex( checksum, modulus ){
 	return mod(checksum, modulus);
 }
 
+function calcSigOfText(text, pem){
+	var signer = crypto.createSign(sha);
+	signer.update(text);
+	var sign   = signer.sign(pem, sigEncoding);
+	var sig    = sign.slice(0,sigLength);
+	return sig;
+}
+
+function calcSigOfShareDetails(shareDetails, articleId, pem){
+	validateStringOrThrow('shareDetails', shareDetails,         detailsRegex);
+	validateStringOrThrow(   'articleId',    articleId, uuidRegexWithHyphens);
+	var sig = calcSigOfText(shareDetails + articleId, pem);
+	validateStringOrThrow( 'sig', sig, sigRegex );
+	return sig;
+}
+
 //------------------------------------------
 // exported functions
 
-function encrypt(userId, articleId, salt, time, tokens, context) {
-	context = context || defaultContext;
+function encrypt(userId, articleId, time, tokens, context, pem) {
 	var timeString   = '' + time;
 	var tokensString = constructMaxTokensString(tokens, maxTokensStringLength);
+	context = context || defaultContext;
+	pem     = pem     || "";
```

Comment:
If no pem is passed, you set the value as an empty string, which would then throw on line 277.